src/Core/AdminConsole/OrganizationFeatures/OrganizationUsers/DeleteManagedOrganizationUserAccountCommand.cs

```diff
@@ -13,87 +14,102 @@ public class DeleteManagedOrganizationUserAccountCommand : IDeleteManagedOrganiz
-        if (userToDelete == null)
-        {
-            throw new NotFoundException("User not found.");
-        }
 
-        await _userService.DeleteAsync(userToDelete);
-        await _eventService.LogOrganizationUserEventAsync(orgUser, EventType.OrganizationUser_Deleted);
+        await RepositoryDeleteUserAsync(organizationId, orgUser, deletingUserId, managementStatus);
     }
 
-    public async Task<IEnumerable<(OrganizationUser, string)>> DeleteManyUsersAsync(Guid organizationId, IEnumerable<Guid> orgUserIds)
+    public async Task<IEnumerable<(Guid, string)>> DeleteManyUsersAsync(Guid organizationId, IEnumerable<Guid> orgUserIds, Guid? deletingUserId)
     {
-        var results = new List<(OrganizationUser, string)>();
+        var results = new List<(Guid, string)>();
         var orgUsers = await _organizationUserRepository.GetManyAsync(orgUserIds);
         var managementStatus = await _getOrganizationUsersManagementStatusQuery.GetUsersOrganizationManagementStatusAsync(organizationId, orgUserIds);
 
-        foreach (var orgUser in orgUsers)
+        foreach (var orgUserId in orgUserIds)
         {
-            if (!orgUser.UserId.HasValue)
-            {
-                results.Add((orgUser, "Invalid organization user."));
-                continue;
-            }
-
-            if (!managementStatus.TryGetValue(orgUser.Id, out var isManaged) || !isManaged)
-            {
-                results.Add((orgUser, "User is not managed by the organization."));
-                continue;
-            }
-
             try
             {
-                var userToDelete = await _userService.GetUserByIdAsync(orgUser.UserId.Value);
-                if (userToDelete == null)
+                var orgUser = orgUsers.FirstOrDefault(ou => ou.Id == orgUserId);
+                if (orgUser == null || orgUser.OrganizationId != organizationId)
                 {
-                    results.Add((orgUser, "User not found."));
-                    continue;
+                    throw new NotFoundException("Organization user not found.");
                 }
 
-                await _userService.DeleteAsync(userToDelete);
-                await _eventService.LogOrganizationUserEventAsync(orgUser, EventType.OrganizationUser_Deleted);
-                results.Add((orgUser, ""));
+                await RepositoryDeleteUserAsync(organizationId, orgUser, deletingUserId, managementStatus);
+                results.Add((orgUserId, ""));
             }
             catch (Exception e)
             {
-                results.Add((orgUser, e.Message));
+                results.Add((orgUserId, e.Message));
             }
         }
 
         return results;
     }
+
+    private async Task RepositoryDeleteUserAsync(Guid organizationId, OrganizationUser orgUser, Guid? deletingUserId, IDictionary<Guid, bool> managementStatus = null)
+    {
+        if (deletingUserId.HasValue && orgUser.UserId.Value == deletingUserId.Value)
```

Comment:
Possible null reference exception on `orgUser.UserId.Value`. You should be able to fix this by moving it below the next `if` block (which checks for `HasValue`).